### PR TITLE
differentiate between atlas config object name and path

### PR DESCRIPTION
**Pull request by tpetracca**
@SerialVelocity 

This is to address issues seen by @cholve in #909

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/920)

### CODE REVIEW

File: atlasdb-service-server/src/main/java/com/palantir/atlasdb/server/AtlasDbServiceServerConfiguration.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
     private final AtlasDbConfig atlasdb;
 
-    public AtlasDbServiceServerConfiguration(@JsonProperty(AtlasDbConfigs.ATLASDB_CONFIG_ROOT) AtlasDbConfig atlasdb) {
+    public AtlasDbServiceServerConfiguration(@JsonProperty(AtlasDbConfigs.ATLASDB_CONFIG_OBJECT_NAME) AtlasDbConfig atlasdb) {
```
Comment (SerialVelocity) [inline atlasdb-service-server/src/main/java/com/palantir/atlasdb/server/AtlasDbServiceServerConfiguration.java]: This seems super weird. This is usually just a hardcoded string. It seems like AtlasDbConfigs should stay exactly the same while this changes to `@JsonProperty("atlasdb")`.

Comment (tpetracca) [inline atlasdb-service-server/src/main/java/com/palantir/atlasdb/server/AtlasDbServiceServerConfiguration.java]: I'm happy to have it be that way.

Comment (tpetracca) [inline atlasdb-service-server/src/main/java/com/palantir/atlasdb/server/AtlasDbServiceServerConfiguration.java]: also if we rename the service to "timelock" or something we should re-hardcode it to that.  Unless this code is already in use somewhere?

Comment (SerialVelocity) [inline atlasdb-service-server/src/main/java/com/palantir/atlasdb/server/AtlasDbServiceServerConfiguration.java]: This is completely unrelated for now. This is what we name the configuration block which currently is called `atlas`, `atlasdb`, etc in products

**Comment by gsheasby** [comment]
This PR hasn't moved for a while - @tpetracca @SerialVelocity should we be merging it in?

**Comment by SerialVelocity** [comment]
I prefer `ATLASDB_CONFIG_ROOT` over `ATLASDB_CONFIG_OBJECT_PATH` but I don't mind us just merging this.

**Comment by tpetracca** [comment]
For clarity the OBJECT_PATH is "/atlasdb" not "atlasdb".  Would you prefer we removed the last commit https://github.com/palantir/atlasdb/pull/920/commits/aed2bb3358595df0808ef1767936df340da485d7

**Comment by SerialVelocity** [comment]
Yes, I would expect `*_OBJECT_PATH`/`*_ROOT`/`*_ROOT_PATH` to be `/atlasdb`, not `atlasdb`. Let's just merge this. :)

**Comment by rhero** [comment]
Hey guys @tpetracca @SerialVelocity - does this affect any product teams? Changing any config things always jumps out, and I didn't see any release notes for this.

**Comment by tpetracca** [comment]
Should have no effect on anyone.  The variable name was confusing and the atlasdbserverserviceconfiguration should not have been referencing it in the first place (was a bug).
